src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -204,7 +205,28 @@ private class TaskLauncherActor(
     case RecheckIfBackOffUntilReached => OfferMatcherRegistration.manageOfferMatcherStatus()
   }
 
+  @SuppressWarnings(Array("all")) // async/await
   private[this] def receiveTaskLaunchNotification: Receive = {
+    case InstanceOpSourceDelegate.InstanceOpRejected(op, TaskLauncherActor.OfferOperationRejectedTimeoutReason) =>
+      // Send a phony task update to trigger a rescheduling.
+      if (inFlightInstanceOperations.exists(_.instanceId == op.instanceId)) {
+        instanceMap.get(op.instanceId).foreach { instance =>
+          import org.apache.mesos.Protos
+          val (taskId, _) = instance.tasksMap.head
```

Comment:
will this work also for pods with multiple tasks? I think that if you change status of one to failed, it won't transition the whole instance to failed

Reply:
It will be in `DEGRADED` status if I'm not mistaken.